src/test/java/uk/gov/companieshouse/api/testdata/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,213 @@
+
+        Users capturedUser = usersCaptor.getValue();
+        assertNotNull(capturedUser.getRoles(), "User roles should not be null");
+        assertEquals(1, capturedUser.getRoles().size(), "User should have one role assigned");
+        assertEquals(capturedRole.getId(), capturedUser.getRoles().get(0), "The assigned role should match the captured role");
+
+        assertNotNull(userTestData.getUserId(), "User ID should not be null");
+        assertNotNull(userTestData.getEmail(), "Email should not be null");
+        assertTrue(userTestData.getForename().contains("Forename"), "Forename should contain Forename");
+        assertTrue(userTestData.getSurname().contains("Surname"), "Surname should contain Surname");
+    }
+    @Test
+    void testUserExists() {
+        when(userRepository.findById("userId")).thenReturn(Optional.of(new Users()));
+        boolean userExists = userServiceImpl.userExists("userId");
+
+        assertTrue(userExists, "User should exist");
+        verify(userRepository, times(1)).findById("userId");
+    }
+
+    @Test
+    void testUserDoesNotExist() {
+        when(userRepository.findById(any(String.class))).thenReturn(Optional.empty());
+
+        boolean userExists = userServiceImpl.userExists("userId");
+
+        assertFalse(userExists, "User should not exist");
+        verify(userRepository, times(1)).findById("userId");
+    }
+
+    @Test
+    void testDeleteUser() throws DataException {
+        Users mockUser = new Users();
+        mockUser.setId("userId");
+
+        when(userRepository.findById("userId")).thenReturn(Optional.of(mockUser));
+        doNothing().when(userRepository).delete(any(Users.class));
+
+        assertTrue(userServiceImpl.userExists("userId"), "User should exist before deletion");
+
+        userServiceImpl.delete("userId");
+
+        verify(userRepository, times(1)).delete(mockUser);
+    }
+
+    @Test
+    void testDeleteUserNotFound() {
+        when(userRepository.findById("userId")).thenReturn(Optional.empty());
+
+        assertFalse(userServiceImpl.userExists("userId"), "User should not exist before deletion");
+
+        assertThrows(DataException.class, () -> userServiceImpl.delete("userId"));
+    }
+
+    @Test
+    void testDeleteUserWithRoles() throws DataException {
+        String userId = "userId";
+        Users mockUser = new Users();
+        mockUser.setId(userId);
+        mockUser.setRoles(List.of("role1", "role2"));
+
+        Roles mockRole1 = new Roles();
+        mockRole1.setId("role1");
+        Roles mockRole2 = new Roles();
+        mockRole2.setId("role2");
+
+        when(userRepository.findById(userId)).thenReturn(Optional.of(mockUser));
+        when(roleRepository.findById("role1")).thenReturn(Optional.of(mockRole1));
+        when(roleRepository.findById("role2")).thenReturn(Optional.of(mockRole2));
+        doNothing().when(roleRepository).delete(any(Roles.class));
+        doNothing().when(userRepository).delete(any(Users.class));
```

Comment:
We don't need these lines as it is a mock object (not a spy) and we are verifying the calls later on

Reply:
removed these line of as its not required